Feature/namespace selector

avihuly opened the pull request.

stakater-user commented:
@avihuly Yikes! You better fix it before anyone else finds out! [Build](https://github.com/stakater/Reloader/commit/edbad456375e627025efb9e1ddf2c14844558ce8/checks) has Failed!

stakater-user commented:
@avihuly Image is available for testing. `docker pull stakater/reloader:SNAPSHOT-PR-356-08e6f81a`

matankila commented:
this feature is what i need!! please merge this PR :)

faizanahmad055 commented:
@avihuly Thank you for submitting the PR. Can you please update the readme which is how we can use the selectors and explain why you choose to use the key-value pairs? I am not able to run and test this locally :)

stakater-user commented:
@avihuly Image is available for testing. `docker pull stakater/reloader:SNAPSHOT-PR-356-481eeeff`

stakater-user commented:
@avihuly Image is available for testing. `docker pull stakater/reloader:SNAPSHOT-PR-356-e702610d`

avihuly commented:
@faizanahmad055  can you take another look please?

stakater-user commented:
@avihuly Image is available for testing. `docker pull stakater/reloader:SNAPSHOT-PR-356-de21a400`

stakater-user commented:
@avihuly Image is available for testing. `docker pull stakater/reloader:SNAPSHOT-PR-356-15d7263c`

avihuly commented:
@faizanahmad055 I fixed the empty namespace labels issue, can you run it now? Additionally, i added [this](https://github.com/stakater/Reloader/pull/356/files#diff-00bfa1a4357de36005018c2d162cb0e8bbe642b9af915a39ab2a74f11e749f42R101) to prevent unnecessary api calls if the `--namespace-selector` arg is not passed

stakater-user commented:
@avihuly Image is available for testing. `docker pull stakater/reloader:SNAPSHOT-PR-356-e7cfafd6`

avihuly commented:
Hi @faizanahmad055,

The issue was that the reloader serviceaccount didn't have the get namespace permissions. 

I Added the necessary verb to the clusterrole in the helm chart under the 'reloader.namespaceSelector' condition.

additional, I added  your log on startup to indicate the `--namespace-selector` flag is on and an error log if the get namespace call fails.

stakater-user commented:
@avihuly Image is available for testing. `docker pull stakater/reloader:SNAPSHOT-PR-356-8ebbb476`

stakater-user commented:
@avihuly Image is available for testing. `docker pull stakater/reloader:SNAPSHOT-PR-356-9e7b7096`
